private/pkg/storage/bucket.go

```diff
@@ -82,6 +85,11 @@ func PutWithChunkSize(sizeInBytes int64) PutOption {
 // The Put operation is complete and the path will be readable once the
 // returned WriteObjectCloser is written and closed (without an error).
 // Any errors will cause the Put to be skipped (no path will be created).
+//
+// TODO: when do we NOT want this to be true? I believe a ton of the Put calls
+// in the buf codebase assume this is true, and were never refactored to take
+// into account that this might not be true. This should likely be removed,
+// and if anything, replaced with a "PutWithoutAtomic".
```

Comment:
This is originally only needed for the CAS code that uses the OS - it helps to guarantee files are written atomically so we never read a partially written file.

The only negative would be if we forget a call to Close() (or there is a kill -9'd process) - in this cases we'd leave behind a temporary file on disk.

When rolling this out, it felt safer to keep it contained to one place to start. I'm ok with it being made the default.

Reply:
Talked about this offline, removed TODOs for now.